bundles/org.openhab.binding.wizlighting/src/main/java/org/openhab/binding/wizlighting/internal/handler/WizLightingHandler.java

```diff
@@ -0,0 +1,688 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.wizlighting.internal.entities.ColorRequestParam;
+import org.openhab.binding.wizlighting.internal.entities.ColorTemperatureRequestParam;
+import org.openhab.binding.wizlighting.internal.entities.DimmingRequestParam;
+import org.openhab.binding.wizlighting.internal.entities.Param;
+import org.openhab.binding.wizlighting.internal.entities.RegistrationRequestParam;
+import org.openhab.binding.wizlighting.internal.entities.SceneRequestParam;
+import org.openhab.binding.wizlighting.internal.entities.SpeedRequestParam;
+import org.openhab.binding.wizlighting.internal.entities.StateRequestParam;
+import org.openhab.binding.wizlighting.internal.entities.SyncResponseParam;
+import org.openhab.binding.wizlighting.internal.entities.SystemConfigResult;
+import org.openhab.binding.wizlighting.internal.entities.WizLightingRequest;
+import org.openhab.binding.wizlighting.internal.entities.WizLightingResponse;
+import org.openhab.binding.wizlighting.internal.enums.WizLightingLightMode;
+import org.openhab.binding.wizlighting.internal.enums.WizLightingMethodType;
+import org.openhab.binding.wizlighting.internal.exceptions.MacAddressNotValidException;
+import org.openhab.binding.wizlighting.internal.utils.ValidationUtils;
+import org.openhab.binding.wizlighting.internal.utils.WizLightingPacketConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WizLightingHandler} is responsible for handling commands, which
+ * are sent to one of the channels.
+ *
+ * @author Sriram Balakrishnan - Initial contribution
+ */
+@NonNullByDefault
+public class WizLightingHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(WizLightingHandler.class);
+
+    private String bulbIpAddress = "bulbIpAddress";
+    private String bulbMacAddress = "bulbMacAddress";
+    private int homeId;
+    private Long updateInterval = DEFAULT_REFRESH_INTERVAL;
+
+    private final WizLightingPacketConverter converter = new WizLightingPacketConverter();
+    private @Nullable ScheduledFuture<?> keepAliveJob;
+    private long latestUpdate = -1;
+    // private RegistrationRequestParam registrationInfo;
+    private int requestId = 0;
+
+    /**
+     * We need to remember some states for increase/decrease commands and scene speed commands
+     */
+    protected Map<String, Integer> stateMap = Collections.synchronizedMap(new HashMap<String, Integer>());
+
+    /**
+     * Default constructor.
+     *
+     * @param thing the thing of the handler.
+     * @throws MacAddressNotValidException if the mac address isn't valid.
+     */
+    public WizLightingHandler(final Thing thing, final RegistrationRequestParam registrationPacket)
+            throws MacAddressNotValidException {
+        super(thing);
+
+        savebulbMacAddressFromConfiguration(this.getConfig());
```

Comment:
Configuration should preferable only be done in the initialize. Also consider using a Configuration class in combination with `getConfigAs` then you don't need to parse the configuration variables, but can directly use them with the type as specificied.

Reply:
Re-done